browser.js

```diff
@@ -98,27 +98,35 @@ function changeTab(next) {
 	$(`a[href$="${pages[ret]}"]`).click();
 }
 
+// sets interval to wait for selector to be ready before firing callback
+function waitFor(selectorString, callback) {
+	const awaitReact = setInterval(() => {
+		if ($(selectorString)) {
+			callback();
+			clearInterval(awaitReact);
+			return;
+		}
+	}, 100);
+}
+
 function newTweet() {
 	if (window.location.pathname.split('/')[1] === 'messages') {
 		$('a[href$="/home"]').click();
-
-		// // TODO: improve the logic here and instead use an
-		// // interval to detect when the button is available
-		setTimeout(newTweet, 400);
-		return;
 	}
 
-	$('a[href$="/compose/tweet"]').click();
+	// wait for new tweet button to click it
+	waitFor('a[href$="/compose/tweet"]', () => {
```

Comment:
Let's make `waitFor` return a Promise instead of a callback. And let the Promise return the DOM element. Then it could just be:

```js
waitFor('a[href$="/compose/tweet"]').then(el => el.click());
```